specification/iotoperations/IoTOperations.Management/models/akri/discoveryHandler.tsp

```diff
@@ -0,0 +1,245 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+using Azure.ResourceManager;
+
+namespace Microsoft.IoTOperations;
+
+/**
+ * AkriDiscoveryHandler resource
+ */
+@added(Versions.`2025-07-01-preview`)
+@parentResource(InstanceResource)
+model AkriDiscoveryHandlerResource
+  is ProxyResource<AkriDiscoveryHandlerProperties> {
+  /**
+   * Name of AkriDiscoveryHandler resource
+   */
+  @pattern("^[a-z0-9][a-z0-9-]*[a-z0-9]$")
+  @key("akriDiscoveryHandlerName")
+  @path
+  @minLength(3)
+  @maxLength(63)
+  @segment("akriDiscoveryHandlers")
+  name: string;
+
+  /**
+   * Edge location of the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property"
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  extendedLocation: ExtendedLocation;
+}
+
+/**
+ * AkriDiscoveryHandler properties
+ */
+// At the moment, the CRD properties are defined in snake_case. We should change them to camelCase to be consistent with the rest of the API.
+@added(Versions.`2025-07-01-preview`)
+model AkriDiscoveryHandlerProperties {
+  /**
+   * The status of the last operation.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /**
+   * Metadata about AIO.
+   */
+  aioMetadata?: AkriDiscoveryHandlerAioMetadata;
+
+  /**
+   * Additional configuration for the AkriDiscoveryHandler.
+   */
+  additionalConfiguration?: string;
```

Comment:
This has been modified to be a map and not a stringified json. It matches the definition in `ConnectorTemplate`.